src/hypothesis/strategies.py

```diff
@@ -836,31 +836,80 @@ def dm_func(denom):
 
 @cacheable
 @defines_strategy
-def decimals(
-    min_value=None, max_value=None, allow_nan=None, allow_infinity=None
-):
-    """Generates instances of decimals.Decimal.
+def decimals(min_value=None, max_value=None,
+             allow_nan=True, allow_infinity=None, places=None):
+    """Generates instances of decimals.Decimal, which may be:
 
-    - If min_value is not None, all values will be >= min_value.
-    - If max_value is not None, all values will be <= max_value.
-    - If min_value or max_value is not None, it is an error to enable
-      allow_nan.
-    - If both min_value and max_value are not None, it is an error to enable
-      allow_infinity.
+    - A finite rational number, between ``min_value`` and ``max_value``.
+    - Not a Number, if ``allow_nan`` is truthy.
```

Comment:
I think it was a matter of principle rather than implementation: NaN isn't *outside* bounds either.  As NaN-handling is an easy source of bugs, and to minimise the interactions between settings, I thought it should be separate.

Easy to change if you disagree, since my scientific background makes me think of NaN as a significant value which maybe *should* be ignored for most cases?